### ISSUE TITLE
Suggestion: add support for msys2

### ISSUE THREAD

**Issue by Kreijstal** [ddnet/ddnet#9437]
Install msys2: 
install pactoys:
```
pacman -Syu git pactoys --needed
```

```
pacboy -S --needed cmake:p curl:p ffmpeg:p freetype:p  glew:p glslang:p gtest:p libnotify:p libpng:p opusfile:p python:p rust:p SDL2:p spirv-tools:p sqlite:p vulkan-headers:p wavpack:p x264:p
```

Unfortunately msys2 doesn't support ssp.
But seems removing the SSP line at 555 on CMakeLists.txt builds DDNet-Server.exe under msys2

**Comment by Kreijstal** [comment]
Saving time typing PKGNAME:p instead of $MINGW_PACKAGE_PREFIX-PKGNAME

> What's the advantage of installing pactoys pacboy when the packages can also be installed with pacman directly?

**Comment by Kreijstal** [comment]
nvm msys2 does support ssp it is under libssp.a and not libssp-0.dll provided by https://packages.msys2.org/packages/mingw-w64-x86_64-crt-git

**Comment by Robyt3** [comment]
> Saving time typing PKGNAME:p instead of $MINGW_PACKAGE_PREFIX-PKGNAME

I prefer not needing another tool for that when it's not necessary. I wouldn't spent time typing this anyway, there's a button to copy the text to the clipboard on GitHub.

**Comment by Kreijstal** [comment]
@Robyt3 Yeah, I guess the real fix is user install deps however they like to install them, and cmake detects if msys2 then it removes the ssp line
